src/tool/hpcrun/gpu/gpu-trace.c

```diff
@@ -378,14 +381,7 @@ gpu_trace_create
 {
   // Init variables
   gpu_trace_t *trace = gpu_trace_alloc();
-
-  // Create a new thread for the stream without libmonitor watching
-  monitor_disable_new_threads();
-
   trace->thread = gpu_trace_demultiplexer_push(trace->trace_channel);
-  atomic_fetch_add(&active_streams_counter, 1);
```

Comment:
You may want to keep `atomic_fetch_add` here instead of the thread that you created.

Though unlikely, it is still possible that by the time the main thread reaches the end, the thread you created is suspended and never executed till `atomic_fetch_add`. In this way the main thread will see `active_streams_counter` as `actual streams - 1`.

Reply:
I did that because it happened that the tracing thread was killed before it started and we had a deadlock because the trace counter is never decremented.

Reply:
How it is killed? I think the tracing thread is only created if there's any trace item. Then the tracing thread should have seen the counter value before you kill it.

Reply:
It happened that libmonitor intercepted pthread_create (even though monitor_disable_new_threads was called before pthread_create) and there called real_pthread_attr_destroy so the thread was never really created (I posted this problem in hpcrun on Slack).  I asked Mark, but he is not sure why that happened. 

As result, I taught it is a good idea to increment and decrement stream counter from the same thread.